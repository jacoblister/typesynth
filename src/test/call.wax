(func callEmptyTest 
  (print "callEmptyTest:")
)

(func callIntTest(param x int) (result int)
  (print "callIntTest:")
  (return x)
)

(func callArrTest (param a (arr int)) (result (arr int))
  (print "callArrTest:")
  (set a 0 4)
  (set a 1 5)
  (set a 2 6)
  (return a)
)

(func callReturnNothing
  (print "callReturnNothing:")
  (return)
)

(func main (result int)
  (call callEmptyTest)
  (print (call callIntTest 2))
  
  (let a (arr int) (alloc (arr int) 1 2 3))
  (let r (arr int) (call callArrTest a))
  (print (get a 0))
  (print (get a 1))
  (print (get a 2))
  (print (get r 0))
  (print (get r 1))
  (print (get r 2))

  (call callReturnNothing)

  (return 0)
)