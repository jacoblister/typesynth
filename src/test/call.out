callEmptyTest:
callIntTest:
2
callArrTest:
4
5
6
4
5
6
callReturnNothing:
